test/Common/TestOrchestrations.cs

```diff
@@ -730,6 +732,98 @@ public static async Task<string> StringStoreWithCreateDelete([OrchestrationTrigg
             return "ok";
         }
 
+        public static async Task<string> CallFaultyEntity([OrchestrationTrigger] IDurableOrchestrationContext ctx)
+        {
+            // construct entity id and proxy
+            var entityId = ctx.GetInput<EntityId>();
+            var entity = ctx.CreateEntityProxy<TestEntityClasses.IFaultyEntity>(entityId);
+            bool isClassBased = entityId.EntityName == "ClassBasedFaultyEntity".ToLowerInvariant();
+            async Task ExpectException(Task t)
+            {
+                if (isClassBased)
+                {
+                    await Assert.ThrowsAsync<TargetInvocationException>(() => entity.SetThenThrow(333));
+                }
+                else
+                {
+                    await Assert.ThrowsAsync<TestEntityClasses.FaultyEntity.SerializableKaboom>(() => entity.SetThenThrow(333));
+                }
+            }

```

Comment:
this needs another case to handle v1 correctly, since v1 throws a FunctionFailedException

Reply:
Is this comment still valid? From what I can tell, we are catching a FunctionFailedException above?